Testscripts/Windows/CPU-Offline.ps1

```diff
@@ -0,0 +1,154 @@
+				Write-LogInfo "CPUOfflineKernelBuild.sh is still running in the VM!"
+			}
+		}
+		if ($vmCount -le 0){
+			Write-LogInfo "CPUOfflineKernelBuild.sh is done successfully"
+		} else {
+			Throw "CPUOfflineKernelBuild.sh didn't finish in the VM!"
+		}
+
+		# ##################################################################################
+		# Reboot VM
+		Write-LogInfo "Rebooting VM!"
+		$TestProvider.RestartAllDeployments($AllVMData)
+
+		# ##################################################################################
+		# Running CPU channel change
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./channel_change.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed channel_change script inside VM"
+
+		# Wait for kernel compilation completion. 60 min timeout
+		$timeout = New-Timespan -Minutes 60
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
+			Wait-Time -seconds 15
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "cat ~/state.txt"
+			if ($state -eq "TestCompleted") {
+				$kernelCompileCompleted = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "cat ~/constants.sh | grep job_completed=0" -runAsSudo
+				if ($kernelCompileCompleted -ne "job_completed=0") {
+					Write-LogErr "channel_change.sh finished on $($VMData.RoleName) but job was not successful!"
+				} else {
+					Write-LogInfo "channel_change.sh finished on $($VMData.RoleName)"
+					$vmCount--
+				}
+				break
+			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "channel_change.sh finished with SKIPPED state!"
+				$resultArr = $resultSkipped
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestFailed") {
+				Write-LogErr "channel_change.sh didn't finish successfully!"
+				$resultArr = $resultFail
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestAborted") {
+				Write-LogInfo "channel_change.sh finished with Aborted state!"
+				$resultArr = $resultAborted
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} else {
+				Write-LogInfo "channel_change.sh is still running in the VM!"
+			}
+		}
+		if ($vmCount -le 0){
+			Write-LogInfo "channel_change.sh is done"
+		} else {
+			Throw "channel_change.sh didn't finish in the VM!"
+		}
+
+		$testResult = $resultPass
+	} catch {
+		$ErrorMessage =  $_.Exception.Message
+		$ErrorLine = $_.InvocationInfo.ScriptLineNumber
+		Write-LogErr "EXCEPTION : $ErrorMessage at line: $ErrorLine"
+	} finally {
+		if (!$testResult) {
+			$testResult = $resultAborted
```

Comment:
Does this script will return Aborted for every exception? Should we expect Failed for runtime exceptions?

Reply:
if testResult does not have the value, then yes.